### PR TITLE
Timeout for kube API watcher

### PR DESCRIPTION
**Pull request by flashvoid**
This commit implemets timeouts on client side,
it fixes current issue with loosing control on server timeouts,
but after each reconnect listener receives request to recreate all policies.

To avoid recreating each and every police on timeouts this commit
introduces PolicyId tag into iptables rules.

### CODE REVIEW

**Comment by jbrendel** [comment]
Are you saying that this patch is also taking care of #262? https://paninetworks.kanbanize.com/ctrl_board/3/cards/262/details

File: NetworkPolicy/k8s-listener.py [hunk 1]
```diff
@@ -258,9 +258,12 @@ def policy_update(romana_address_scheme, policy_definition, delete_policy=False)
 
     """
 
+    # PolicyId is a uniq tag that we are going to use to check if rule is applied already
+    policy_id      = "%s/%s" % ( policy_definition['policy_name'], policy_definition['policy_namespace'] )
```
Comment (jbrendel) [inline NetworkPolicy/k8s-listener.py:262]: Minor point: We don't do long lines in Python (80 chars is good).

File: NetworkPolicy/k8s-listener.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def make_rules(addr_scheme, policy_def):
                                            (u32_in_match, in_chain_name)),
         _make_rule(policy_chain_name, '-m u32 --u32 "%s" -j %s' %
                                            (u32_out_match, out_chain_name)),
-        _make_rule(policy_chain_name, '-j RETURN')
+        _make_rule(policy_chain_name, '-m comment --comment PolicyId=%s -j RETURN' % policy_id)
```
Comment (jbrendel) [inline NetworkPolicy/k8s-listener.py:117]: I noticed that the comment is only applied to some rules?

Comment (jbrendel) [inline NetworkPolicy/k8s-listener.py:117]: Hm. Too late, I'll have to look at it tomorrow.

Comment (flashvoid) [inline NetworkPolicy/k8s-listener.py:117]: We only need it in one rule.

Comment (jbrendel) [inline NetworkPolicy/k8s-listener.py:117]: Ok, got it now. Sorry, it was a bit too late last night. You are just looking for this comment to see if this particular policy has been put into the iptables yet, therefore you'll only need one of those. Ok.

**Comment by jbrendel** [comment]
Looks good.

**Pull request merged**